src/components/tags/Tag.jsx

```diff
@@ -1,15 +1,33 @@
-import { React, useContext } from "react";
+import { React, useContext, useEffect } from "react";
 import SelectedTagContext from "../../context/SelectedTagContext";
+import { useSearchParams } from "react-router-dom";
 
 function Tag({ title, id }) {
   const { setSelectedTag, selectedTag } = useContext(SelectedTagContext);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  useEffect(() => {
+    if (searchParams) {
+      if (searchParams.get("tag") === "null" || !searchParams.get("tag")) {
+        setSelectedTag(null);
+      } else {
+        setSelectedTag(Number(searchParams.get("tag")));
```

Comment:
I would use a variable to make the code easier to read (in my opinion)

```suggestion
      const tag = searchParams.get("tag")
      setSelectedTag((tag === "null" || !tag) ? null : Number(tag))
```

or maybe

```suggestion
      const id = Number(searchParams.get("tag"))
      setSelectedTag((isNaN(id) || id < 1) ? null : id)
```